jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/DataCloudConnection.java

```diff
@@ -539,12 +549,25 @@ public String getSchema() {
     @Override
     public void abort(Executor executor) {}
 
+    /**
+     * Set the network timeout for network operations in this connection. This is a safety net to protect against hanging connections.
+     * To enforce a query timeout rather use {@link DataCloudStatement#setQueryTimeout(int)}.
+     * A too low network timeout might cause the JDBC driver to fail to operate properly.
+     * @param executor This will be ignored
+     * @param milliseconds The network timeout in milliseconds.
+     */
     @Override
-    public void setNetworkTimeout(Executor executor, int milliseconds) {}
+    public void setNetworkTimeout(Executor executor, int milliseconds) {
+        networkTimeout = Duration.ofMillis(milliseconds);
+    }
```

Comment:
I would prefer if the user can pass this in directly, as a strictly typed `Duration`.

Also, why do we not accept this as part of `ConnectionProperties`? Might be useful to pass this in via the JDBC property bag, too, e.g. from DBeaver for quick experiments?

```suggestion
    public void setNetworkTimeout(Executor executor, Duration timeout) {
        networkTimeout = timeout;
    }
```